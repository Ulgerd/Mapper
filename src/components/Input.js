import React, {useState} from 'react';
import { connect } from 'react-redux';
import { setNewDot } from '../actions/rootActions.js'
import { strToCoordsReq } from '../utils/request.js'
import { formatDotData } from '../utils/formatDotData.js'
import styled from 'styled-components'

const StyledInput = styled.input`
  width: 100%;
`
export const onEnter = async (e, location, setInputVal, setNewDot) => {
  if (e.key === 'Enter' && location !== '') {
    setInputVal('')
    let locData = await strToCoordsReq(location);
    if (locData.length > 0) {
      let formData = formatDotData(locData[0].GeoObject);
      setNewDot(formData)
    } else {
      alert("По вашему запросу ничего не найдено")
    };
  }
}

export function Input(props) {
  const [inputVal, setInputVal] = useState('');
  return (
      <div>
        <StyledInput
          onChange = {(e) => setInputVal(e.target.value)}
          value={inputVal}
          onKeyPress={(e) => onEnter(e, inputVal, setInputVal, props.setNewDot)}
        />
      </div>
  );
}

const mapDispatchToProps = dispatch => ({
  setNewDot: (formData) => {dispatch(setNewDot(formData))},
})

export default connect(null, mapDispatchToProps) (Input);
