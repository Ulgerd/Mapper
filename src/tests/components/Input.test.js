import React from 'react';
import { mount } from 'enzyme'
import { Input } from '../../components/Input.js';
import renderer from 'react-test-renderer';

describe('Input component tests',()=>{
    let wrapper

    beforeEach(()=>{
        wrapper = mount(<Input />)
    })

    it('render the DUMB component', () => {
       expect(wrapper.length).toEqual(1)
    });

    it('contains div', () => {
        expect(wrapper.find('div')).toHaveLength(1)
    });

    it('contains input', () => {
        expect(wrapper.find('input')).toHaveLength(1)
    });

    it('renders correctly', () => {
      const tree = renderer
        .create(<Input />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
});
