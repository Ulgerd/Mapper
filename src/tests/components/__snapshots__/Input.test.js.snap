// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component tests renders correctly 1`] = `
<div>
  <input
    className="sc-bdVaJa UoSJW"
    onChange={[Function]}
    onKeyPress={[Function]}
    value=""
  />
</div>
`;
